src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -81,7 +81,8 @@ export class HttpConnection implements IConnection {
         this.baseUrl = this.resolveUrl(url);
 
         options = options || {};
-        options.logMessageContent = options.logMessageContent || false;
+        options.logMessageContent = options.logMessageContent === undefined ? false : options.logMessageContent;
+        options.withCredentials = options.withCredentials === undefined ? true : options.withCredentials;
```

Comment:
Considering we later do `request.withCredentials === true` in one place and pass through `withCredentials` value directly in others, we should probably try to force truthy values to actually be `true` here to ensure no difference between xhr and fetch.

Reply:
> we should probably try to force truthy values to actually be `true` here

Isn't that what I'm doing here?

Reply:
Even if `withCredentials !== undefined`, that doesn't necessarily mean that it's `true` or `false`, right?

Reply:
So if someone set `withCredentials = new Error();`? I'm not sure if we want to protect from people doing silly things like that.

Reply:
Protect, no. I don't mind immediately throwing in that case. The risk is allowing it and interpreting it differently in different places.